app/views/providers/legal_aid_applications/_proceeding_type.html.erb

```diff
@@ -0,0 +1,24 @@
+<div id="<%= proceeding_type.code %>" class="govuk-grid-row proceeding-item">
+  <div class="govuk-grid-column-two-thirds text-column">
+    <div class="govuk-caption-m">
+      <%= proceeding_type.ccms_category_law %> &gt; <%= proceeding_type.ccms_matter %>
+    </div>
+    <span class="govuk-body"><%= proceeding_type.meaning %></span>
+  </div>
+
+  <%= form_for([:providers, @legal_aid_application]) do |form| %>
+  <div class="govuk-grid-column-one-third control-column">
+    <%= hidden_field_tag :proceeding_type, proceeding_type.code %>
+    <div class="govuk-!-padding-bottom-2"></div>
+    <%= form.submit 'Select and continue', class: 'govuk-button proceeding-link' %>
+  </div>
+  <% end %>
+</div>
+
+<script>
+  <%#
+    Hidding div as soon as possible to minimize appearance on page transition.
+    `raw ... to_json` to keep erblint happy
+  %>
+  $(<%= raw "##{proceeding_type.code}".to_json %>).hide();
```

Comment:
❓ Is the intention here to achieve something functionally equivalent to a:

```js
$( document ).ready(function() {
    $('.proceeding-item').hide();
});

Reply:
Yes and no. The two do the same thing - hide the proceeding item elements. My first version had exactly that code (`$('.proceeding-item').hide();`) at the start of `app/assets/javascripts/proceeding_types.js`. 

However, there is an issue in that it can take a fraction of a second for the items to be hidden after they are rendered, and on Chrome this is visible as a flash of green on page load.

This code is an attempt to minimise this effect. It has two effects: first, it hides each element as soon as it is rendered (rather than after the whole page has rendered). Second, it ensures there is only ever one element visible on the page, so the size of the "green" block is minimised.

If I'm honest, it's not a particularly elegant bit of code and I'd welcome a better suggestion. It's made more ugly by erblint insisting that any ruby code rendered into a `<script>` block must have `to_json` called on it.